tests/dt.go

```diff
@@ -0,0 +1,234 @@
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	RackLabel = "rack"
+	RackNum   = 3
+)
+
+// RegionInfo records detail region info for api usage.
+type RegionInfo struct {
+	ID          uint64              `json:"id"`
+	StartKey    string              `json:"start_key"`
+	EndKey      string              `json:"end_key"`
+	RegionEpoch *metapb.RegionEpoch `json:"epoch,omitempty"`
+	Peers       []*metapb.Peer      `json:"peers,omitempty"`
+
+	Leader          *metapb.Peer      `json:"leader,omitempty"`
+	DownPeers       []*pdpb.PeerStats `json:"down_peers,omitempty"`
+	PendingPeers    []*metapb.Peer    `json:"pending_peers,omitempty"`
+	WrittenBytes    uint64            `json:"written_bytes,omitempty"`
+	ReadBytes       uint64            `json:"read_bytes,omitempty"`
+	ApproximateSize int64             `json:"approximate_size,omitempty"`
+	ApproximateKeys int64             `json:"approximate_keys,omitempty"`
+}
+
+// RegionsInfo contains some regions with the detailed region info.
+type RegionsInfo struct {
+	Count   int           `json:"count"`
+	Regions []*RegionInfo `json:"regions"`
+}
+
+func (oa *operatorActions) LabelNodes() error {
+	nodes, err := oa.kubeCli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for i, node := range nodes.Items {
+		index := i % RackNum
+		node.Labels[RackLabel] = fmt.Sprintf("rack%d", index)
+		oa.kubeCli.CoreV1().Nodes().Update(&node)
+	}
+	return nil
+}
+
+func (oa *operatorActions) LabelNodesOrDie() {
+	err := oa.LabelNodes()
+	if err != nil {
+		slack.NotifyAndPanic(err)
+	}
+}
+
+func (oa *operatorActions) CheckDisasterTolerance(cluster *TidbClusterConfig) error {
+	nodeMap := map[string]corev1.Node{}
+	nodes, err := oa.kubeCli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes.Items {
+		nodeMap[node.Name] = node
+	}
+
+	pds, err := oa.kubeCli.CoreV1().Pods(cluster.Namespace).List(
+		metav1.ListOptions{LabelSelector: labels.SelectorFromSet(
+			label.New().Instance(cluster.ClusterName).PD().Labels(),
+		).String()})
+	if err != nil {
+		return err
+	}
+	err = oa.checkDisasterTolerance(pds.Items, nodeMap)
+	if err != nil {
+		return err
+	}
+
+	tikvs, err := oa.kubeCli.CoreV1().Pods(cluster.Namespace).List(
+		metav1.ListOptions{LabelSelector: labels.SelectorFromSet(
+			label.New().Instance(cluster.ClusterName).TiKV().Labels(),
+		).String()})
+	if err != nil {
+		return err
+	}
+	err = oa.checkDisasterTolerance(tikvs.Items, nodeMap)
+	if err != nil {
+		return err
+	}
+
+	tidbs, err := oa.kubeCli.CoreV1().Pods(cluster.Namespace).List(
+		metav1.ListOptions{LabelSelector: labels.SelectorFromSet(
+			label.New().Instance(cluster.ClusterName).TiDB().Labels(),
+		).String()})
+	if err != nil {
+		return err
+	}
+	return oa.checkDisasterTolerance(tidbs.Items, nodeMap)
+}
+
+func (oa *operatorActions) checkDisasterTolerance(allPods []corev1.Pod, nodeMap map[string]corev1.Node) error {
```

Comment:
`revive` warns about `function/method name only differs in capitalization`, I thinks we should avoid this.

Reply:
good, got it